### ISSUE TITLE
Opt-out Race Condition in 3.0.0.beta

### ISSUE THREAD

**Issue by nibe** [mixpanel/mixpanel-swift#458]
The following code should always send the `User Opted Out` event. But more often than not it does not send the event. Maybe using the completion handler in `flush` it would work but nobody likes callback hell.
```swift
Mixpanel.mainInstance().registerSuperProperties(["Opt-out": true])
Mixpanel.mainInstance().track(event: "User Opted Out")
Mixpanel.mainInstance().flush()
Mixpanel.mainInstance().optOutTracking()
```

The following works more often but is also unreliable. When it does not work, your own `$opt_in` event is missing, too. And there is no callback to blame here.
```swift
Mixpanel.mainInstance().optInTracking()
Mixpanel.mainInstance().registerSuperProperties(["Opt-out": false])
Mixpanel.mainInstance().track(event: "User Opted In")
```

In the Amplitude iOS SDK both of these cases work reliably.

Branch 3.0.0.beta
Compiled for macOS
macOS 11.4
Xcode 12.4

**Comment by zihejia** [comment]
hi @nibe, we will have a look, thanks.

**Comment by zihejia** [comment]
Hi @nibe , the issue has now been fixed in [v2.10.2](https://github.com/mixpanel/mixpanel-swift/releases/tag/v2.10.2). Again, thank you so much for bringing it up. I'm closing it now, please feel free to reopen it if needed.

**Comment by zihejia** [comment]
Updated in 3.0.0.beta branch as well.
